File: app/src/main/java/nl/acidcats/teatimer/TeaTimeApplication.java
package nl.acidcats.teatimer;

import android.app.Application;
import android.app.NotificationManager;
import android.os.Build;

import com.facebook.stetho.Stetho;
import com.pixplicity.easyprefs.library.Prefs;

import nl.acidcats.teatimer.util.NotificationUtil;
import timber.log.Timber;

/**
 * Application class for the TeaTimer application
 */

public class TeaTimeApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        Prefs.initPrefs(this);

        Timber.plant(new Timber.DebugTree());
        if (BuildConfig.DEBUG) {
            Stetho.initializeWithDefaults(this);
        }

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationUtil.createNotificationChannel(this,
                    R.string.notification_channel_id, R.string.notification_channel_name, R.string.notification_channel_description,
                    NotificationManager.IMPORTANCE_DEFAULT, false, false);

            NotificationUtil.createNotificationChannel(this,
                    R.string.alarm_channel_id, R.string.alarm_channel_name, R.string.alarm_channel_description,
                    NotificationManager.IMPORTANCE_HIGH, true, true);
        }
    }
}

File: app/src/main/java/nl/acidcats/teatimer/TeaTimeApplication.kt
package nl.acidcats.teatimer

import android.app.Application
import com.github.ajalt.timberkt.Timber
import com.github.ajalt.timberkt.Timber.DebugTree
import nl.acidcats.teatimer.di.helperModule
import nl.acidcats.teatimer.helpers.NotificationHelper
import org.koin.android.ext.android.getKoin
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin

/**
 * Application class for the TeaTimer application
 */

class TeaTimeApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        Timber.plant(DebugTree())

        startKoin {
            androidLogger()
            androidContext(this@TeaTimeApplication)
            koin.loadModules(listOf(helperModule))
        }

        initNotificationChannels()
    }

    private fun initNotificationChannels() {
        val helper = getKoin().get<NotificationHelper>()
        helper.createNotificationChannel(
            channelIdId = R.string.notification_channel_id,
            nameId = R.string.notification_channel_name,
            descId = R.string.notification_channel_description,
            isImportant = false,
            enableVibration = false,
            enableLights = false
        )

        helper.createNotificationChannel(
            channelIdId = R.string.alarm_channel_id,
            nameId = R.string.alarm_channel_name,
            descId = R.string.alarm_channel_description,
            isImportant = true,
            enableVibration = true,
            enableLights = true
        )
    }
}
